src/libs/SelectionScraper/index.ts

```diff
@@ -64,17 +66,17 @@ const getHTMLOfSelection = () => {
             // and finally commonAncestorContainer.parentNode.closest('data-testid') is targeted dom.
             if (range.commonAncestorContainer instanceof HTMLElement) {
                 parent = range.commonAncestorContainer.closest(`[${tagAttribute}]`);
-            } else {
-                parent = range.commonAncestorContainer.parentNode.closest(`[${tagAttribute}]`);
+            } else if (range.commonAncestorContainer.parentNode) {
+                parent = (range.commonAncestorContainer.parentNode as HTMLElement).closest(`[${tagAttribute}]`);
             }
 
             // Keep traversing up to clone all parents with 'data-testid' attribute.
             while (parent) {
                 const cloned = parent.cloneNode();
                 cloned.appendChild(child);
-                child = cloned;
+                child = cloned as DocumentFragment;
 
-                parent = parent.parentNode.closest(`[${tagAttribute}]`);
+                parent = (parent.parentNode as HTMLElement).closest(`[${tagAttribute}]`);
```

Comment:
```suggestion
                parent = (parent.parentNode as HTMLElement | null)?.closest(`[${tagAttribute}]`) ?? null;
```

`parentNode` can be null.

Reply:
Good catch, thanks!